### PR TITLE
8330205: Initial troff manpage generation for JDK 24

### PR DESCRIPTION
**Pull request by pavelrappo**
Please review this mechanical change to man pages. This PR should be integrated after https://github.com/openjdk/jdk/pull/18787.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330205](https://bugs.openjdk.org/browse/JDK-8330205): Initial troff manpage generation for JDK 24 (**Sub-task** - P4)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**)
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19119/head:pull/19119` \
`$ git checkout pull/19119`

Update a local copy of the PR: \
`$ git checkout pull/19119` \
`$ git pull https://git.openjdk.org/jdk.git pull/19119/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19119`

View PR using the GUI difftool: \
`$ git pr show -t 19119`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19119.diff">https://git.openjdk.org/jdk/pull/19119.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19119#issuecomment-2098236985)

### CODE REVIEW

**Comment by pavelrappo** [comment]
This PR is standalone as opposed to dependent because the https://github.com/openjdk/jdk/pull/18787 dependency currently has a merge conflict, which complicates the required workflow.

This comment serves as a reminder to merge master into this PR once https://github.com/openjdk/jdk/pull/18787 has been integrated. If we don't do that, Skara will likely create a backport issue:

> (⚠️ The fixVersion in this issue is [24] but the fixVersion in .jcheck/conf is 23, a new backport will be created when this pr is integrated.)

**Review by jddarcy — APPROVED** [review]

**Comment by jddarcy** [comment]
> This PR is standalone as opposed to dependent because the #18787 dependency currently has a merge conflict, which complicates the required workflow.
> 
> This comment serves as a reminder to merge master into this PR once #18787 has been integrated. If we don't do that, Skara will likely create a backport issue:
> 
> > (⚠️ The fixVersion in this issue is [24] but the fixVersion in .jcheck/conf is 23, a new backport will be created when this pr is integrated.)

Thanks @pavelrappo. For as a small risk-reduction exercise, the fixVersion on the bug could also be changed to tbd, but if it is integrated after the main start-of-release PR it should be fine.

**Comment by pavelrappo** [comment]
Thanks for reviewing it Joe, I'm now delegating integration of this PR to @JesperIRL, you, or anyone who will be integrating https://github.com/openjdk/jdk/pull/18787.

**Review by jonathan-gibbons — APPROVED** [review]

File: src/java.base/share/man/java.1 [hunk 1]
```diff
@@ -3853,7 +3853,7 @@ This option was deprecated in JDK 16 by \f[B]JEP 396\f[R]
 403\f[R] [https://openjdk.org/jeps/403].
 .SH REMOVED JAVA OPTIONS
 .PP
-These \f[V]java\f[R] options have been removed in JDK 23 and using them
+These \f[V]java\f[R] options have been removed in JDK 24 and using them
```
Comment (dholmes-ora) [inline src/java.base/share/man/java.1:3856]: This is incorrect. You can't just change 23 to 24 here as the actual set of flags listed below will be will be different.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
I think we may have to restore this to a standalone start-of-release change done after the other start-of-release changes. I had forgotten about the need to update the flags sections of java.1

File: src/java.base/share/man/java.1 (hunk 1 above)
Comment (pavelrappo) [reply src/java.base/share/man/java.1:3856]: David, my understanding is that you agree that this is okay (per our out-of-band, follow-up discussion).

**Comment by pavelrappo** [comment]
Withdrawing this, as a different approach is required.

**Pull request closed without merging**